### PR TITLE
remove curry code as PoC puzzle hash never changes

### LINKED ISSUE #2: refactor new puzzle hash calculation
- [ ] [`curry_hashes`](https://github.com/Chia-Network/chia-blockchain/blob/eab43eb80096dffd81388e1e1224c4517977c7bd/chia/wallet/puzzles/curry.clib#L84) instead of [`puzzle-hash-of-curried-function`](https://github.com/Chia-Network/chia-blockchain/blob/eab43eb80096dffd81388e1e1224c4517977c7bd/chia/wallet/puzzles/curry-and-treehash.clib#L78)
- [ ] Consider having the partial coin be double-curried: the first curry includes all things that don't change from coin to coin (pretty much everything you're currying in this case - OFFER_MOD, CAT_MOD, RATE, TAIL_HASH, MAKER_PK, etc.) to produce a new puzzle. Then, second curry has only one arg, SELF_HASH - which represents the hash of the mod after the 1st curry. So, to re-create, you can just do `(puzzle-hash-of-curried-function SELF_HASH (sha256 1 puzzle-hash-of-curried-function )`

### PR DESCRIPTION
**Pull request by kimsk**
Actually, we could remove puzzle hash calculation, `curry_hashes` or `puzzle-hash-of-curried-function`, https://github.com/dexie-space/partial-cli-beta/issues/2, in our puzzle totally as we can curry-in `CAT_OFFER_HASH`, https://github.com/dexie-space/partial-cli-beta/issues/4 and the puzzle hash of the partial offer coin is never changed once it's created because all curried parameters never change.

```lisp
MOD_HASH                ; self puzzle hash 
FEE_PH                  ; puzzle hash that will receive the fee
FEE_RATE                ; fee rate (0-10000), e.g., 1% is represented as 100
MAKER_PH                ; maker puzzle hash (both receive and clawback)
MAKER_PK                ; maker public key used in clawback case
TAIL_HASH               ; request CAT tail hash
CAT_OFFER_HASH          ; puzzle hash of CAT settlement coin
RATE                    ; number of CAT per XCH, e.g., 1 XCH = 100 CATs, rate = 100000
```

However, this approach need us to continue using `RATE`, https://github.com/dexie-space/partial-cli-beta/issues/8.

Pros:
- resolve https://github.com/dexie-space/partial-cli-beta/issues/2 and https://github.com/dexie-space/partial-cli-beta/issues/4
- The cost of PoC coin in taking case reduces significantly ~2/3.

Cons:
- Can't do https://github.com/dexie-space/partial-cli-beta/issues/8
- Will this work for CAT-CAT trading?

### CODE REVIEW

File: partial_cli/puzzles/partial.clsp [hunk 1]
```diff
@@ -5,43 +5,46 @@
         MAKER_PH                ; maker puzzle hash (both receive and clawback)
         MAKER_PK                ; maker public key used in clawback case
         TAIL_HASH               ; request CAT tail hash
+        CAT_OFFER_HASH          ; puzzle hash of CAT settlement coin
```
Comment (kimsk) [inline partial_cli/puzzles/partial.clsp:8]: new curried parameter.

File: partial_cli/puzzles/partial.clsp [hunk 2]
```diff
@@ -5,43 +5,46 @@
...
         RATE                    ; number of CAT per XCH, e.g., 1 XCH = 100 CATs, rate = 100000
-        partial_coin_amount     ; amount of partial offer coin
-        coin_id                 ; coin id of the partial offer coin
+        my_amount               ; amount of partial offer coin
+        my_id                   ; coin id of the partial offer coin
+        my_puzzle_hash          ; puzzle hash of the partial offer coin
```
Comment (kimsk) [inline partial_cli/puzzles/partial.clsp:12]: new parameter.

File: partial_cli/puzzles/fns.clsp [hunk 3]
```diff
@@ -38,52 +42,24 @@
...
                     )
-                )
-                (cat_offer_mod_hash
-                    (puzzle-hash-of-curried-function 
-                        CAT_MOD_HASH
-                        OFFER_MOD_HASH
-                        (sha256 ONE TAIL_HASH)
-                        (sha256 ONE CAT_MOD_HASH)
-                    )
-                )
```
Comment (kimsk) [inline partial_cli/puzzles/fns.clsp:58]: `cat_offer_mod_hash` is never changed once PoC is created.

File: partial_cli/puzzles/fns.clsp [hunk 4]
```diff
@@ -38,52 +42,24 @@
...
-                        (sha256 ONE RATE) 
-                        (sha256 ONE TAIL_HASH)
-                        (sha256 ONE MAKER_PK)
-                        (sha256 ONE MAKER_PH)
-                        (sha256 ONE FEE_RATE)
-                        (sha256 ONE FEE_PH)
-                        (sha256 ONE MOD_HASH)
-                    )
-                )
-            )
```
Comment (kimsk) [inline partial_cli/puzzles/fns.clsp:83]: The puzzle hash is never changed once PoC is created.

File: partial_cli/puzzles/partial.clsp [hunk 5]
```diff
@@ -5,43 +5,46 @@
...
+                            (list ASSERT_MY_COIN_ID my_id)
+                            (assert-taker-announcement MAKER_PH CAT_OFFER_HASH my_id request_cat_mojos)
                             (create-settlement-coin (- taken_mojos_or_clawback fee_mojos))
                             (list CREATE_COIN FEE_PH fee_mojos) 
                         )
                         ; if new_partial_coin_amount is greater than 0, create a new partial coin
                         (if (> new_partial_coin_amount 0)
                             (list
-                                (create-partial-coin MOD_HASH FEE_PH FEE_RATE MAKER_PH MAKER_PK TAIL_HASH RATE new_partial_coin_amount)
+                                (list ASSERT_MY_PUZZLEHASH my_puzzle_hash)
```
Comment (kimsk) [inline partial_cli/puzzles/partial.clsp]: New condition when the new PoC is created.

File: partial_cli/puzzles/partial.clsp (hunk 2 above)
Comment (kimsk) [reply partial_cli/puzzles/partial.clsp:12]: also resolve https://github.com/dexie-space/partial-cli-beta/issues/7

**Comment by denisu** [comment]
I really like the lower cost. If we can deal with the precision in `RATE` and for `CAT-CAT` trades then I am good with it.

**Pull request merged**